sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2119,10 +2119,13 @@ object GpuOverrides {
     expr[First](
       "first aggregate operator",
       ExprChecks.aggNotWindow(
-        TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_64,
+        (TypeSig.psNote(TypeEnum.STRUCT, "only supports in GroupBy aggregation") +
+          TypeSig.psNote(TypeEnum.ARRAY, "only supports in GroupBy aggregation") +
```

Comment:
I'm not sure why we need a partial support note here?  Does STRUCT and ARRAY not work for first/last on a reduction?  If so then we need tests to verify that we will fall back to the CPU correctly in those cases, and a follow on issue to fix it in cudf.

Reply:
Done.